### PR TITLE
fix: ensure files can only be picked once

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
In current design, sst files may be picked multiple times.

## Detailed Changes
- Mark files as in compacting when pick files candidates, and reset it to false when CompactionTask is dropped.

## Test Plan
Manually

### CODE REVIEW

File: analytic_engine/src/compaction/mod.rs [hunk 1]
```diff
@@ -348,18 +361,59 @@ impl CompactionTask {
...
+        self.inputs.is_empty()
+    }
+
+    pub fn expired(&self) -> &[ExpiredFiles] {
+        &self.expired
+    }
+
+    pub fn inputs(&self) -> &[CompactionInputFiles] {
+        &self.inputs
+    }
```
[the remark this answers is not in the document]
Comment (jiacai2050) [reply analytic_engine/src/compaction/mod.rs]: fixed

File: analytic_engine/src/table/version.rs [hunk 2]
```diff
@@ -727,7 +727,9 @@ impl TableVersion {
         picker_ctx: PickerContext,
         picker: &CompactionPickerRef,
     ) -> picker::Result<CompactionTask> {
-        let inner = self.inner.read().unwrap();
+        // Pick will set FileHandle to being_compacted state, so we require
+        // write lock here to prevent other threads pick same file to compact.
+        let inner = self.inner.write().unwrap();
 
         picker.pick_compaction(picker_ctx, &inner.levels_controller)
```
Comment (Rachelint) [inline analytic_engine/src/table/version.rs:734]: "levels_controller: &LevelsController" -> 
"levels_controller: &mut LevelsController"

Force to use wirte lock here?

Comment (jiacai2050) [reply analytic_engine/src/table/version.rs:734]: Good suggestion.

File: analytic_engine/src/compaction/mod.rs [hunk 3]
```diff
@@ -318,13 +318,26 @@ pub struct ExpiredFiles {
 
 #[derive(Default, Clone)]
 pub struct CompactionTask {
-    pub compaction_inputs: Vec<CompactionInputFiles>,
-    pub expired: Vec<ExpiredFiles>,
+    inputs: Vec<CompactionInputFiles>,
+    expired: Vec<ExpiredFiles>,
```
[the remark this answers is not in the document]
Comment (jiacai2050) [reply analytic_engine/src/compaction/mod.rs:322]: This sounds strange.

**Pull request merged**